docs/docsite/rst/network/dev_guide/developing_resource_modules_network.rst

```diff
@@ -375,3 +375,90 @@ The tests rely on a role generated by the resource module builder. After changes
                    -e structure=role \
                    -e model=models/myos/interfaces/myos_interfaces.yml \
                    site.yml
+
+
+.. _testing_resource_modules:
+
+
+Unit testing Ansible network resource modules
+=============================================
+
+
+This section walks through an example of how to develop unit tests for Ansible network resource
+modules.
+
+See :ref:`testing_units` and :ref:`testing_units_modules` for general documentation on Ansible unit tests for modules.
+Please read those pages first to understand unit tests and why and when you should use them.
+
+.. note::
+
+   The structure of the unit tests matches
+   the structure of the code base, so the tests that reside in the :file:`test/units/modules/network` directory
+   are organized by module groups.
+
+Using mock objects to unit test Ansible network resource modules
+----------------------------------------------------------------
+
+
+Mock objects (from https://docs.python.org/3/library/unittest.mock.html) can be very
+useful in building unit tests for special or difficult cases, but they can also
+lead to complex and confusing coding situations.  One good use for mocks would be to
+simulate an API. The ``mock`` Python package is bundled with Ansible (use
+``import units.compat.mock``).
+
+You can mock the device connection and output from the device as follows:
+
+.. code-block:: python
+
+   self.mock_get_config = patch('ansible.module_utils.network.common.network.Config.get_config')
+   self.get_config = self.mock_get_config.start()
+
+   self.mock_load_config = patch('ansible.module_utils.network.common.network.Config.load_config')
+   self.load_config = self.mock_load_config.start()
+
+   self.mock_get_resource_connection_config = patch('ansible.module_utils.network.common.cfg.base.get_resource_connection')
+   self.get_resource_connection_config = self.mock_get_resource_connection_config.start()
+
+   self.mock_get_resource_connection_facts = patch('ansible.module_utils.network.common.facts.facts.get_resource_connection')
+   self.get_resource_connection_facts = self.mock_get_resource_connection_facts.start()
+
+   self.mock_edit_config = patch('ansible.module_utils.network.eos.providers.providers.CliProvider.edit_config')
+   self.edit_config = self.mock_edit_config.start()
+
+   self.mock_execute_show_command = patch('ansible.module_utils.network.eos.facts.l2_interfaces.l2_interfaces.L2_interfacesFacts.get_device_data')
+   self.execute_show_command = self.mock_execute_show_command.start()
+
+The facts file of the module now includes a new method, ``get_device_data``. Call ``get_device_data`` here to emulate the device output.
+
+
+Mocking out device data
```

Comment:
I found this confusing on a first read, because I wasn't sure whether "mocking out device data" meant "adding fake device data for testing purposes" or "removing device data for testing purposes". Since we use "mocking X" without the "out" above, let's use the verb that way throughout.

```suggestion
Mocking device data
```